pytorch_lightning/accelerators/accelerator_connector.py

```diff
@@ -282,22 +375,47 @@ def select_training_type_plugin(self):
             plugin = ddp_plugin_cls(
                 parallel_devices=self.parallel_devices,
                 num_nodes=self.num_nodes,
-                cluster_environment=cluster_environment,
+                cluster_environment=self.cluster_environment,
                 sync_batchnorm=self.sync_batchnorm,
             )
         elif self.use_dp:
             plugin = DataParallelPlugin(parallel_devices=self.parallel_devices)
         elif self.use_horovod:
             plugin = HorovodPlugin(parallel_devices=self.parallel_devices)
         elif self.on_tpu:
-            plugin = SingleTPUPlugin(self.tpu_id)
+            if isinstance(self.tpu_cores, list):
+                plugin = SingleTPUPlugin(self.tpu_id)
```

Comment:
Is this correct? Shouldn't we use spawn for multiple cores?

Reply:
yes this is correct. And yes, for multiple cores, we use spawn.

For TPU the currently supported modes are to either pass a number of cores as an int or to pass a list with a single core id (we only need the list here to differentiate core_id 2 from 2 cores)